snippets: Add example of cryptographic code (#6)
bjorn3/inline_asm_catalogue: snippets/faz-donenfeld-neves_crypto.md

* Authors: Armando Faz, Jason A. Donenfeld, Samuel Neves
* Reason: side-channel resistance required + performance critical
* Source: https://github.com/armfazh/rfc7748_precomputed

> * License: [BSD-3-clause](https://github.com/armfazh/rfc7748_precomputed/blob/master/LICENSE)

Reviewer: Maybe add `* Language: C`, as the rest of the snippets in this repo are written in Rust.

# Code

> I did not write this code myself, but I feel this snippet is a good example of

Reviewer: ```suggestion
This snippet is probably a good example of
```

Using `I` can be confusing as to who it refers to.
